src/toil/test/jobStores/jobStoreTest.py

```diff
@@ -1339,6 +1339,56 @@ def testOverlargeJob(self):
         self.assertEqual(jobsInJobStore, [str(overlargeJob)])
         jobstore.delete(overlargeJob.jobStoreID)
 
+    def testMultithreadImportFile(self) -> None:
+        """ Tests that importFile is thread-safe."""
+
+        from toil.jobStores.abstractJobStore import AbstractJobStore
+        from concurrent.futures.thread import ThreadPoolExecutor
+        import multiprocessing
+        import timeit
+
+        threads = (
+            2,
+            multiprocessing.cpu_count(),  # maximum
```

Comment:
We have a Toil-specific CPU counting method (`toil.lib.threading.cpu_count`) that understands Kubernetes pods and won't make as many threads as there are cores on the underlying host.